### PR TITLE
[DOCS-6680] Add keyword dictionary and priority level

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

Adds info on SDS keyword dictionary and priority level.
Moves the regex content to the custom rules section.
Move the OOTB section up.

[DOCS-6680](https://datadoghq.atlassian.net/browse/DOCS-6680)

### Merge instructions

- [x] Please merge after reviewing

### Additional notes

[DOCS-6680]: https://datadoghq.atlassian.net/browse/DOCS-6680?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/sensitive_data_scanner.md [hunk 1]
```diff
@@ -62,10 +52,23 @@ Sensitive Data Scanner supports Perl Compatible RegEx (PCRE), but the following
...
+    - Conditional patterns
+    - Backtracking control verbs
+    - The \C "single-byte" directive (which breaks UTF-8 sequences)
+    - The \R newline match
+    - The \K start of match reset directive
+    - Callouts and embedded code
+    - Atomic grouping and possessive quantifiers
 - **Define scope:** Specify whether you want to scan the entire event or just specific attributes. You can also choose to exclude specific attributes from the scan.
-- **Add tags:** Specify the tags you want to associate with events where the values match the specified regex pattern. Datadog recommends using `sensitive_data` and `sensitive_data_category` tags. These tags can then be used in searches, dashboards, and monitors.
+- **Create keyword dictionary**: Add keywords to provide additional context when matching regex conditions. For example, if you are scanning for a nine-digit credit card number, you can add keywords like `account number`, `bank card`, and `cc #` to refine the match.
```
Comment (maycmlee) [inline content/en/sensitive_data_scanner.md:68]:
```suggestion
- **Create keyword dictionary**: Add keywords to provide additional context when matching regex conditions. For example, if you are scanning for a nine-digit Visa credit card number, you can add keywords like `visa`, `credit`, and `card` to refine the match.
```

File: content/en/sensitive_data_scanner.md [hunk 2]
```diff
@@ -42,18 +42,8 @@ Sensitive Data Scanner can be found under [Organization Settings][1].
...
-  - Arbitrary zero-width assertions
-  - Subroutine references and recursive patterns
-  - Conditional patterns
-  - Backtracking control verbs
-  - The \C "single-byte" directive (which breaks UTF-8 sequences)
-  - The \R newline match
-  - The \K start of match reset directive
-  - Callouts and embedded code
-  - Atomic grouping and possessive quantifiers
+- **Create keyword dictionary**: Add keywords to provide additional context when matching regex conditions. For example, if you are scanning for a nine-digit credit card number, you can add keywords like `account number`, `bank card`, and `cc #` to refine the match.
```
Comment (maycmlee) [inline content/en/sensitive_data_scanner.md:45]:
```suggestion
- **Create keyword dictionary**: Add keywords to provide additional context when matching regex conditions. For example, if you are scanning for a nine-digit Visa credit card number, you can add keywords like `visa`, `credit`, and `card` to refine the match.
```

File: content/en/sensitive_data_scanner.md [hunk 3]
```diff
@@ -28,49 +28,59 @@ Sensitive data, such as credit card numbers, bank routing numbers, and API keys
...
+    - Backtracking control verbs
+    - The \C "single-byte" directive (which breaks UTF-8 sequences)
+    - The \R newline match
+    - The \K start of match reset directive
+    - Callouts and embedded code
+    - Atomic grouping and possessive quantifiers
+   
+#### Define rule target and action
+
+- **Create keyword dictionary**: Add keywords to tune detection accuracy when matching regex conditions. For example, if you are scanning for a sixteen-digit Visa credit card number, you can add keywords like `visa`, `credit`, and `card` and require that these keywords must be within a specified number of characters of a match. By default, keywords must be within 30 characters before a matched value.
```
[the remark this answers is not in the document]
Comment (buraizu) [reply content/en/sensitive_data_scanner.md:74]: > By default, keywords must be within 30 characters before a matched value.

I'm not totally sure what that means, is that something a user would be expected to understand here?

File: content/en/sensitive_data_scanner.md [hunk 4]
```diff
@@ -28,49 +28,59 @@ Sensitive data, such as credit card numbers, bank routing numbers, and API keys
...
 - **Define scope:** Specify whether you want to scan the entire event or just specific attributes. You can also choose to exclude specific attributes from the scan.
-- **Add tags:** Specify the tags you want to associate with events where the values match the specified regex pattern. Datadog recommends using `sensitive_data` and `sensitive_data_category` tags. These tags can then be used in searches, dashboards, and monitors.
 - **Process matching values:** Optionally, specify whether you want to redact, partially redact, or hash matching values. When redacting, specify placeholder text to replace the matching values with. When partially redacting, specify the position (start/end) and length (# of characters) to redact within matching values. Redaction, partial redaction, and hashing are all irreversible actions.
+- **Add tags:** Specify the tags you want to associate with events where the values match the specified regex pattern. Datadog recommends using `sensitive_data` and `sensitive_data_category` tags. These tags can then be used in searches, dashboards, and monitors.
+- **Set priority level**: Set the priority level for a rule based on your business needs.
```
Comment (buraizu) [inline content/en/sensitive_data_scanner.md:78]:
```suggestion
- **Set priority level:** Set the priority level for a rule based on your business needs.
```

**Review by buraizu — APPROVED** [review]
LGTM overall and approving; just posted a couple of extremely minor updates for consistency and a question about a part that wasn't clear to me.

File: content/en/sensitive_data_scanner.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/sensitive_data_scanner.md:74]: Thanks @buraizu! Yes, I think the user would understand... it's also in the UI: https://a.cl.ly/yAuBXNA9

**Pull request merged**